keps/sig-docs/20191020-doc-policies-for-third-party-content.md

```diff
@@ -0,0 +1,162 @@
+---
+title: doc-policies-for-third-party-content
+authors:
+  - "@jimangel"
+  - "@sftim"
+  - "@zacharysarah"
+owning-sig: sig-docs
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-10-20
+last-updated: 2019-10-20
+status: provisional
+---
+
+# doc-policies-for-third-party-content
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1 (fictional)](#story-1-fictional)
+    - [Story 2 (fictional)](#story-2-fictional)
+    - [Story 3 (actual)](#story-3-actual)
+    - [Story 4 (actual)](#story-4-actual)
+    - [Story 5 (actual)](#story-5-actual)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Graduation Criteria](#graduation-criteria)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+*Note* that this KEP does not target any release; SIG Docs follows a continuous release process for website content.
+
+## Summary
+
+Feature docs are not a place for vendor pitches. Nevertheless, SIG Docs has seen pull requests that are more or less a request to place advertising content on the Kubernetes website.
+
+Feature docs also contain content hosted in multiple places (“dual-sourced content”). A good practice for code project docs is to host single-sourced content only, and to provide links to other providers’ single-sourced content.
+This KEP defines a policy on documentation content, so that authors can judge what is appropriate to propose and so that PR approvers can make consistent, fair decisions during the review process.
+
+## Motivation
+
+SIG Docs publishes Kubernetes documentation on kubernetes.io, in line with its [charter](https://github.com/kubernetes/community/blob/master/sig-docs/charter.md#scope), and also sets standards for website content. Prior to this KEP there were no clear guidelines or standards showing what documentation is acceptable.
+
+The Kubernetes documentation is currently a mix of both 1) documentation describing the Kubernetes open source project; and 2) content describing how to install or use Kubernetes on several third party Kubernetes offerings.
+
+Some third party content is necessary in order for Kubernetes to function. For example, Docker, networking policy (CNI plugins), Ingress controllers, and logging all require third party components.
+Pages like [Logging Using Elasticsearch and Kibana](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/) are highly specific to a third party offering and seem more like third party product documentation than Kubernetes open source documentation.
```

Comment:
AIUI, this paragraph aims to include:
- Docker
- CNI plugins
- Ingress controllers
- logging helpers

as places where 3rd party components are useful.

https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/ seems like the sort of page where every vendor for a log service might reasonable want to put up a similar page about how to hook your cluster up to that logging service. This starts to look like vendor pitch territory.

@timothysc do you have some alternative wording in mind?

Reply:
> seems like the sort of page where every vendor for a log service might reasonable want to put up a similar page about how to hook your cluster up to that logging service. This starts to look like vendor pitch territory. 

This statement is a false choice ^.  If there is a community set of best practices that use OSS tools, and may or may not have a private backer, we should provide documentation on what we consider to be best practices using the public facing artifacts.  That said, we should remove private, proprietary integrations and point to external documentation.

Reply:
@timothysc Thanks for the review. I have questions.

One of the goals of this KEP is to achieve specific processes for how these things _actually get done_, not just by hand waving.

```
1. Who decides when to include third-party content?
1. What standard of quality and review must be met before docs include third-party content?
```

> We vet specific versions so there is a case by case basis for the changes.

When you say "we", who is that specifically? What is the specific vetting process?

> If there is a community set of best practices that use OSS tools, and may or may not have a private backer, **we should provide documentation on what we consider to be best practices** using the public facing artifacts. (emphasis added)

That is the motivation for this KEP. This is the place where we create specific practices for Kubernetes.

> This example is not a good one IMO. 

It's a real example in the docs that we have to deal with. How should we proceed?

> These are each OSS projects whose integration provides value to the community. 

Who specifically determines whether integrations provide value to the community and how?

> If they have a page related to kubernetes then it makes sense to not host. 

If projects have default permission to host on kubernetes.io, what is the incentive for projects to host their own documentation?

```
1. When should the Kubernetes documentation host third party documents that we are not the source/authority of?
1. How does the Kubernetes project handle third party content that is not kept up to date or hosts?
```

> However, in the early days of kubernetes, k8s was the plan of record.

Can you point to where that's documented? Is there a good reason for tradition to continue simply because [_that's the way we've always done it_](https://www.forbes.com/sites/forbeslacouncil/2019/01/28/the-most-dangerous-phrase-in-business-weve-always-done-it-this-way/#2c7c02140f7e)?

Reply:
The question of Elasticsearch is particularly compelling, given the [tension](https://news.ycombinator.com/item?id=19359602) surrounding it. I raise the example specifically because it isn't clear who vets these inclusions or how&mdash;and good docs need these processes to be clear and transparent, not just a matter of a single developer saying, "It's there because I said so."

Reply:
That seems odd.  I would expect the authoring sig would own and maintain.  In this case I would expect sig-apps to maintain this work.